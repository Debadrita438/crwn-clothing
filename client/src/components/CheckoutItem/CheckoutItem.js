import { connect } from 'react-redux';
import { addItem, clearItemFromCart, removeItem } from '../../redux/cart/cartAction';

import { 
    CheckoutItemContainer, 
    ImageContainer, 
    TextContainer, 
    QuantityStyles, 
    RemoveButtonStyles 
} from './CheckoutItem.styles';

const mapDispatchToProps = dispatch => ({
    clearItem: item => dispatch(clearItemFromCart(item)),
    addItem: item => dispatch(addItem(item)),
    removeItem: item => dispatch(removeItem(item))
});

const CheckoutItem = ({ cartItem, clearItem, addItem, removeItem }) => {
    const { name, imageUrl, price, quantity } = cartItem;
    return ( 
        <CheckoutItemContainer>
            <ImageContainer>
                <img src={ imageUrl } alt="item" />
            </ImageContainer>
            <TextContainer>{ name }</TextContainer>
            <QuantityStyles>
                <div onClick = {() => removeItem(cartItem)}>&#10094;</div>
                <span>{ quantity }</span>
                <div onClick = {() => addItem(cartItem)}>&#10095;</div>
            </QuantityStyles>
            <TextContainer>{ price }</TextContainer>
            <RemoveButtonStyles onClick = {() => clearItem(cartItem)}>&#10008;</RemoveButtonStyles>
        </CheckoutItemContainer>
    );
}
 
export default connect(null, mapDispatchToProps)(CheckoutItem);