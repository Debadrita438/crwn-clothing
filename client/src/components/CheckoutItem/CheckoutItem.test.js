import { shallow } from 'enzyme';

import { CheckoutItem } from './CheckoutItem';

describe('CheckoutItem Component', () => {
    let wrapper;
    let mockClearItem;
    let mockAddItem;
    let mockRemoveItem;

    beforeEach(() => {
        mockClearItem = jest.fn();
        mockAddItem = jest.fn();
        mockRemoveItem = jest.fn();

        const mockProps = {
            cartItem: {
                imageUrl: 'www.testImage.com',
                price: 10,
                name: 'sneakers',
                quantity: 2
            },
            clearItem: mockClearItem,
            addItem: mockAddItem,
            removeItem: mockRemoveItem
        }

        wrapper = shallow(<CheckoutItem { ...mockProps } />);
    });


    it('should render the CheckoutItem component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the removeItem method when left arrow is clicked', () => {
        wrapper.find('QuantityStyles').childAt(0).simulate('click');
        expect(mockRemoveItem).toHaveBeenCalled();
    });

    it('should render the addItem method when right arrow is clicked', () => {
        wrapper.find('QuantityStyles').childAt(2).simulate('click');
        expect(mockAddItem).toHaveBeenCalled();
    });

    it('should render the clearItem method when cross is clicked', () => {
        wrapper.find('RemoveButtonStyles').simulate('click');
        expect(mockClearItem).toHaveBeenCalled();
    });
});