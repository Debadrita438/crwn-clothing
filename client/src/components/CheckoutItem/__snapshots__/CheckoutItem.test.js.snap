// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem Component should render the CheckoutItem component 1`] = `ShallowWrapper {}`;
